tesseract-server/src/handlers/util.rs

```diff
@@ -70,7 +70,11 @@ pub fn generate_source_data(cube: &Cube) -> SourceMetadata {
     }
 }
 
-
+pub fn get_user_auth_level(req: &HttpRequest<AppState>) -> Option<i32> {
+    let jwt_secret = &req.state().env_vars.jwt_secret;
+    let user_token = extract_token(req);
+    user_auth_level(jwt_secret, &user_token)
+}
 
 pub fn verify_api_key(req: &HttpRequest<AppState>, cube: &Cube) -> Result<(), HttpResponse> {
```

Comment:
Just to confirm, we want to keep the old check for public cubes AND also add the new `min_auth_level` check? I had initially thought that we would only be using the latter, but I guess it makes sense to keep the public flag for applications that won't use JWTs